pkg/obfuscate/json.go

```diff
@@ -139,45 +144,45 @@ func (p *jsonObfuscator) obfuscate(data []byte) (string, error) {
 				out.WriteByte('"')
 				out.WriteString(result)
 				out.WriteByte('"')
-				p.transformingValue = false
+				st.transformingValue = false
 				valBuf = valBuf[:0]
-			} else if p.keeping && depth < p.keepDepth {
-				p.keeping = false
+			} else if st.keeping && depth < st.keepDepth {
+				st.keeping = false
 			}
 
 		case scanBeginLiteral, scanContinue:
 			// starting or continuing a literal
-			if p.transformingValue {
+			if st.transformingValue {
 				valBuf = append(valBuf, c)
 				continue
-			} else if p.key {
+			} else if st.key {
 				// it's a key
 				keyBuf = append(keyBuf, c)
-			} else if !p.keeping {
+			} else if !st.keeping {
 				// it's a value we're not keeping
-				if !p.wiped {
+				if !st.wiped {
 					out.Write([]byte(`"?"`))
-					p.wiped = true
+					st.wiped = true
 				}
 				continue
 			}
 
 		case scanObjectKey:
 			// done scanning key
-			k := strings.Trim(string(keyBuf), `"`)
-			if !p.keeping && p.keepKeys[k] {
+			k := string(bytes.Trim(keyBuf, `"`))
```

Comment:
I'm surprised this makes things faster! Any docs or posts that led you to discover this? I took a quick look at the two implementations of bytes.Trim and strings.Trim and they seemed pretty similar in performance?

Reply:
@ajgajg1134 It really reduces one allocation in some cases. I think it's because there is some escaping value in `strings.Trim` but not in `bytes.Trim`.